web/src/lib/user_auth.ts

```diff
@@ -0,0 +1,37 @@
+import { api, UserResponseDto } from '@api';
+import { goto } from '$app/navigation';
+import { get } from 'svelte/store';
+import { session } from '$app/stores';
+
+export async function checkUserAuthStatus(): Promise<UserResponseDto>  {
+    const user = get(session).user;
```

Comment:
You can use `$session` to get the latest value of session `$` is for reactive in Svelte

Ref: https://svelte.dev/tutorial/auto-subscriptions

Reply:
That seems to work only in .svelte files and not in .ts.

Reply:
Ah that is correct